apps/prairielearn/src/pages/instructorStudentDetail/components/StudentAuditEventsTable.tsx

```diff
@@ -0,0 +1,71 @@
+import { FriendlyDate } from '../../../components/FriendlyDate.js';
+import { type StaffAuditEvent } from '../../../lib/client/safe-db-types.js';
+import { type SupportedActionsForTable } from '../../../models/audit-event.types.js';
+
+interface StudentAuditEventsTableProps {
+  events: StaffAuditEvent[];
+}
+
+function renderEnrollmentEventText(event: StaffAuditEvent): string {
+  const { action_detail } = event;
+
+  if (!action_detail) {
+    return 'Enrollment event';
+  }
+
+  const detailMap: Record<SupportedActionsForTable<'enrollments'>, string> = {
+    implicit_joined: 'Enrolled in course',
+    explicit_joined: 'Enrolled in course',
+    invited: 'Invited to course',
+    invitation_accepted: 'Accepted invitation',
+    invitation_rejected: 'Rejected invitation',
+    blocked: 'Blocked from course',
+    unblocked: 'Unblocked from course, now enrolled',
+    invitation_deleted: 'Invitation cancelled', // You can never actually see this state.
+  };
+
+  const detail = detailMap[action_detail as SupportedActionsForTable<'enrollments'>];
+  if (!detail) {
+    throw new Error(`Unknown action detail: ${action_detail}`);
```

Comment:
Throwing an error and killing the entire page in this case is probably bad? At least in PT, we make a best-effort attempt to render unknown events. You're already sort of doing the same with the `!action_detail` case above.

Reply:
If you can think of a scenario, I'll handle this case.